tests/test_templates/test_run_azure_template.py

```diff
@@ -408,6 +414,7 @@ def test_deprovision(template_runner: AzureTemplateRunner):
     template_runner._check_terraform_installation = MagicMock()
     template_runner._check_matcha_directory_exists = MagicMock()
     template_runner._destroy_terraform = MagicMock()
+    template_runner._write_outputs_state_cloud_only = MagicMock()
```

Comment:
Do we want to test whether the existing state file been updated with the expected data?

Reply:
Added